### ISSUE TITLE
Error with 17633EPA Lucille Strong after adding staff time for Don Kirk

### ISSUE THREAD

**Issue by caripang** [babybunny/rebuildingtogethercaptain#345]
Cari created a new staff position yesterday for Don Kirk.

After Greg tried to enter Don's time in, he started receiving an INTERNAL_SERVER error.  Worse I can no longer access that site in ROOMS.  Can the errant record be removed so that we can at least access this site?  And potentially sort out what the issue is with Don's staff record?

https://rebuildingtogethercaptain.appspot.com/room/site/view/5790526028644352/
Internal Server Error

The server has either erred or is incapable of performing the requested operation.

Traceback (most recent call last):
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/webapp2-2.5.2/webapp2.py", line 1535, in __call__
    rv = self.handle_exception(request, response, e)
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/webapp2-2.5.2/webapp2.py", line 1529, in __call__
    rv = self.router.dispatch(request, response)
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/webapp2-2.5.2/webapp2.py", line 1278, in default_dispatcher
    return route.handler_adapter(request, response)
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/webapp2-2.5.2/webapp2.py", line 1102, in __call__
    return handler.dispatch()
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/staff.py", line 56, in dispatch
    super(StaffHandler, self).dispatch(*a, **k)
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/webapp2-2.5.2/webapp2.py", line 572, in dispatch
    return self.handle_exception(e, self.app.debug)
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/webapp2-2.5.2/webapp2.py", line 570, in dispatch
    return method(*args, **kwargs)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/staff.py", line 135, in get
    return common.Respond(self.request, 'site_list_one', d)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/common.py", line 192, in Respond
    request_handler.response.out.write(template.render(params))
  File "/base/data/home/runtimes/python27_experiment/python27_lib/versions/third_party/jinja2-2.6/jinja2/environment.py", line 894, in render
    return self.environment.handle_exception(exc_info, True)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/templates/site_list_one.html", line 1, in top-level template code
    {%extends "base.html"%}
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/templates/base.html", line 55, in top-level template code
    {% block body %}Body Placeholder{% endblock %}
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/templates/site_list_one.html", line 33, in block "body"
    {% include "site_view.html" %}
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/templates/site_view.html", line 267, in top-level template code
    (${{ "%0.2f" % site.StaffTimeTotal() }})
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/ndb_models.py", line 463, in StaffTimeTotal
    return sum(cr.Total() or 0 for cr in self.StaffTimes)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/ndb_models.py", line 463, in <genexpr>
    return sum(cr.Total() or 0 for cr in self.StaffTimes)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/ndb_models.py", line 987, in Total
    return self.HoursTotal() + self.MileageTotal()
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/ndb_models.py", line 975, in HoursTotal
    return self.hours * self.position.get().GetHourlyRate(self.activity_date)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/ndb_models.py", line 826, in GetHourlyRate
    return _GetRateFromArray(self.hourly_rate, self.hourly_rate_after_date, activity_date)
  File "/base/data/home/apps/s~rebuildingtogethercaptain-hrd/issue337.404177349547254627/room/ndb_models.py", line 786, in _GetRateFromArray
    if activity_date_str < dr[0]:
IndexError: list index out of range

**Comment by babybunny** [comment]
P

**Comment by caripang** [comment]
Unfortunately, I think this is related to item #289 since Don is the only new person I've added since Greg (which was involved in affecting item #289).   

I went back in and tried to fix the issue with Don's record and blew up "17000DEL DELETE THIS SITE" and then I compared when Don's & Greg's records showed and made a couple more changes and blew up "17000 Test Site None" so I'm now afraid to try and make any more changes.  

There do not appear to be any issues with any of the other staff positions being inputted.  I just tested adding mileage and an hour of time with Chris's position to 17000RWC with no issues.
